packages/react-native/ReactCommon/react/renderer/core/ComponentDescriptor.h

```diff
@@ -133,7 +136,7 @@ class ComponentDescriptor {
 
   EventDispatcher::Weak eventDispatcher_;
   std::shared_ptr<const ContextContainer> contextContainer_;
-  RawPropsParser rawPropsParser_{};
+  std::unique_ptr<RawPropsParser> rawPropsParser_;
```

Comment:
Can we keep this a direct object, and use the move constructor to optionally pass in a custom instance instead of switching to unique_ptr?